ambry-server/src/main/java/com.github.ambry.server/StatsManager.java

```diff
@@ -60,21 +63,28 @@
   /**
    * Constructs a {@link StatsManager}.
    * @param storageManager the {@link StorageManager} to be used to fetch the {@link Store}s
-   * @param partitionIds a {@link List} of {@link PartitionId}s that are going to be fetched
+   * @param replicaIds a {@link List} of {@link ReplicaId}s that are going to be fetched
    * @param registry the {@link MetricRegistry} to be used for {@link StatsManagerMetrics}
    * @param config the {@link StatsManagerConfig} to be used to configure the output file path and publish period
    * @param time the {@link Time} instance to be used for reporting
    * @throws IOException
    */
-  StatsManager(StorageManager storageManager, List<PartitionId> partitionIds, MetricRegistry registry,
+  StatsManager(StorageManager storageManager, List<? extends ReplicaId> replicaIds, MetricRegistry registry,
       StatsManagerConfig config, Time time) throws IOException {
     this.storageManager = storageManager;
-    totalPartitionIds = partitionIds;
     statsOutputFile = new File(config.outputFilePath);
     publishPeriodInSecs = config.publishPeriodInSecs;
     initialDelayInSecs = config.initialDelayUpperBoundInSecs;
     metrics = new StatsManagerMetrics(registry);
     mapper.setVisibilityChecker(mapper.getVisibilityChecker().withFieldVisibility(JsonAutoDetect.Visibility.ANY));
+    totalPartitionIds = new ArrayList<>();
+    partitionToReplicaMap = new HashMap<>();
+    PartitionId partitionId;
+    for (ReplicaId replicaId : replicaIds) {
+      partitionId = replicaId.getPartitionId();
+      totalPartitionIds.add(partitionId);
+      partitionToReplicaMap.put(partitionId, replicaId);
```

Comment:
Why is `totalPartitionIds` needed? Could you just use `partitionToReplicaMap.keySet()` to iterate through partitions?

If you only need the map you can just do
```java
    partitionToReplicaMap =
        replicaIds.stream().collect(Collectors.toMap(ReplicaId::getPartitionId, Function.identity()));
```

Reply:
Thanks for your suggestion. Will remove `totalPartitionIds` to make code more concise.